components/script/dom/readablestream.rs

```diff
@@ -531,9 +532,13 @@ impl ReadableStreamMethods for ReadableStream {
                 Ok(ConversionResult::Success(val)) => val,
                 Ok(ConversionResult::Failure(error)) => return Err(Error::Type(error.to_string())),
                 _ => {
+                    // Note: given that `JsUnderlyingSource::new` can throw,
+                    // we must clear the exception before returning an error,
+                    // because the constructor will throw in that case.
+                    unsafe { JS_ClearPendingException(*cx) };
```

Comment:
We should return the JSFailed error instead. That avoids throwing anything and leaves the existing exception intact.

Reply:
thank you!